src/SourceGenerators/Uno.UI.SourceGenerators/HotRestart/HotRestartGenerator.cs

```diff
@@ -46,7 +46,14 @@ public void Run()
 			}
 
 			var mauiAppDelegate = _context.Compilation.GetTypeByMetadataName("Microsoft.Maui.MauiUIApplicationDelegate");
-			var shouldDefineAppDelegate = (mauiAppDelegate is null).ToString().ToLowerInvariant();
+
+			if (mauiAppDelegate is not null)
+			{
+				// Support for hot restart is disabled, since the app cannot be started from
+				// the MauiUIApplicationDelegate as the CreateMaui override cannot be implemented
```

Comment:
Should we log a info/warning telling the user that this feature may not work?

Reply:
The generator is probably not a good place to do it, though it may be interesting to add that as an msbuild target. Assuming there's an identifyable maui property (such as `UseMAUI`) we can probably show a message when `IsHotRestartBuild` is set. 

I'll make a separate PR for that, thanks for the suggestion!